File: litfass-server/src/main/kotlin/lit/fass/litfass/server/script/ScriptEngine.kt
package lit.fass.litfass.server.script

/**
 * @author Michael Mair
 */
interface ScriptEngine {
    fun isApplicable(language: ScriptLanguage): Boolean
    fun invoke(script: String, data: Collection<Map<String, Any?>>): Collection<Map<String, Any?>>
}

File: litfass-server/src/main/scala/lit/fass/server/script/ScriptEngine.scala
package lit.fass.server.script

import lit.fass.server.ScriptLanguage.ScriptLanguage
import lit.fass.server.model.DataList

/**
 * @author Michael Mair
 */
trait ScriptEngine {
  def isApplicable(language: ScriptLanguage): Boolean

  def invoke(script: String, data: DataList): DataList
}
